components/dialog/dialog.js

```diff
@@ -149,7 +150,7 @@ class Dialog extends LocalizeCoreElement(AsyncContainerMixin(DialogMixin(LitElem
 						<d2l-button-icon icon="tier1:close-small" text="${this.localize('components.dialog.close')}" @click="${this._abort}"></d2l-button-icon>
 					</div>
 				</div>
-				<div class="d2l-dialog-content" @pending-state="${this._handleAsyncItemState}">${content}</div>
+				<d2l-heading-stack class="d2l-dialog-content" override-level="3" @pending-state="${this._handleAsyncItemState}" style="display: block;">${content}</d2l-heading-stack>
```

Comment:
So just to confirm, this is how a consumer would render the heading?

Reply:
No, this is an example of needing to "reset" the heading stack, since when dialogs open they blank out the rest of the page and right above here render a `<h2>` -- which is why this is resetting things such that if there's a nested heading inside the dialog, it'll be a `h3`.

There's an example of consuming the heading level inside `heading-stack-test.js`, but as discussed in there it's a bit awkward to do. So we're considering also adding a `<d2l-heading>` control that would reduce the complexity there.

Reply:
No, the `d2l-heading-stack` just initializes or increments the heading level variable for that DOM subtree. Inside the stack, they would either have to render an `<h#>` using the controller to resolve the current level as Dave did below in `heading-stack-test.js`, or we'd provide a `<d2l-heading>` component renders the actual heading and hides the gnarly switch statement away.

Reply:
Apologies. I shoulda refreshed the page before submitting my reply.